### PR TITLE
BearTest.py: Add test_download_cached_file

### PR DESCRIPTION
**Pull request by Mixih**
### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
- [x] I have followed the [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference).
- [x] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### Reviewers

Since download_cached_file wasn't covered anymore,
added a test to cover it.

Fixes https://github.com/coala/coala/issues/3110

### CODE REVIEW

File: tests/bears/BearTest.py [hunk 1]
```diff
@@ -179,3 +180,14 @@ def test_new_result(self):
         result = bear.new_result('test message', '/tmp/testy')
         expected = Result.from_values(bear, 'test message', '/tmp/testy')
         self.assertEqual(result, expected)
+
+    def test_download_cached_file(self):
+        url = 'ftp://ftp.gnu.org/gnu/alive/alive-2.0.2.tar.xz'
```
Comment (Nosferatul) [inline tests/bears/BearTest.py]: you should use another url maybe? I see that is a `403 Forbidden`

File: tests/bears/BearTest.py [hunk 2]
```diff
@@ -179,3 +180,14 @@ def test_new_result(self):
         result = bear.new_result('test message', '/tmp/testy')
         expected = Result.from_values(bear, 'test message', '/tmp/testy')
         self.assertEqual(result, expected)
+
+    def test_download_cached_file(self):
+        url = 'https://google.com'
+        filename = 'google.html'
+        uut = TestBear(self.settings, None)
```
Comment (userzimmermann) [inline tests/bears/BearTest.py]: what means `uut`? :)

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3111/commits/ed1b095428585c46b6655f091bb77e8c765ee1d7

**Comment by sils** [comment]
@rultor merge

File: tests/bears/BearTest.py (hunk 2 above)
Comment (Mixih) [inline tests/bears/BearTest.py]: unit under test? Is this incorrect usage?

File: tests/bears/BearTest.py [hunk 3]
```diff
@@ -179,3 +180,14 @@ def test_new_result(self):
...
+        result_filename = uut.download_cached_file(url, filename)
+        expected_filename = join(uut.data_dir, filename)
+        self.assertEqual(result_filename, expected_filename)
+        result_filename = uut.download_cached_file(url, filename)
+        self.assertEqual(result_filename, expected_filename)
```
Comment (userzimmermann) [inline tests/bears/BearTest.py]: How do you know that the file wasn't downloaded again? It is also not checked if the file really exists. And if it wasn't there before the first download.

File: tests/bears/BearTest.py (hunk 2 above)
Comment (userzimmermann) [inline tests/bears/BearTest.py]: now that I know it it's correct ;)

File: tests/bears/BearTest.py (hunk 3 above)
Comment (Mixih) [inline tests/bears/BearTest.py]: > It is also not checked if the file really exists
trivial implementation. We can check it if you want. I thought it wasn't completely necessary

Comment (userzimmermann) [inline tests/bears/BearTest.py]: > exists where?

at `result_filename`

Comment (userzimmermann) [inline tests/bears/BearTest.py]: `os.path.exists`

**Pull request merged**